commands/publish_dafsa.py

```diff
@@ -0,0 +1,114 @@
+import requests
+from requests.exceptions import HTTPError
+import tempfile
+import subprocess
+from kinto_http import Client, KintoException
+import sys
+
+COMMIT_HASH_URL = "https://api.github.com/repos/publicsuffix/list/commits/master"
+
+LIST_URL = (
+    "https://raw.githubusercontent.com/publicsuffix/list/master/public_suffix_list.dat"
+)
+
+MAKE_DAFSA_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/make_dafsa.py"
+PREPARE_TLDS_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/prepare_tlds.py"
+
+
+SERVER = "https://kinto.dev.mozaws.net/v1"
+CREDENTIALS = ("arpit73", "s3cr3t")  # (username, password)
+
+
+def handle_request_errors(func):
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+        else:
+            print("Fetching Failed")
+
+    return inner
+
+
+@handle_request_errors
+def get_latest_hash(url):
+    response = requests.get(url)
+    return response.json()["sha"]
+
+
+@handle_request_errors
+def download_resources(*urls, **kwargs):
+    for url in urls:
+        # file_location is found by appending the file_name(at the end of url string) to temp directory
+        file_location = kwargs["directory"] + "/" + url.split("/")[-1]
+        print(file_location)
+        response = requests.get(url, stream=True)
```

Comment:
shouldn't we do something like `response.raise_for_status()` ?

Reply:
Is this for error handling? I'm raising HTTP errors with the `handle_errors()` decorator function.
here https://github.com/arpit73/remote-settings-lambdas/blob/2d798e1cc08da0d395cab1cbd9541cab2191a229/commands/publish_dafsa.py#L22-L33

Reply:
you don't raise errors at all, you catch them and print them. Read the requests docs to see what raise_for_status() does